src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -319,15 +324,67 @@ public static class AkkaClusterHostingExtensions
             {
                 switch (builder.ActorRefProvider.Value)
                 {
-                    case ProviderSelection.Cluster :
-                    case ProviderSelection.Custom :
+                    case ProviderSelection.Cluster:
+                    case ProviderSelection.Custom:
                         return hoconBuilder; // no-op
                 }
             }
 
             return hoconBuilder.WithActorRefProvider(ProviderSelection.Cluster.Instance);
         }
 
+        private static ExtractEntityId ToExtractEntityId(this IMessageExtractor self)
```

Comment:
Ported this from Akka.NET itself in order to make it easy to flip between `IMessageExtractor` and the delegates supported by Akka.Cluster.Sharding.